plugins/askrene/askrene.c

```diff
@@ -275,7 +361,8 @@ static const char *get_routes(const tal_t *ctx,
 			} else {
 				/* Handled below, after other layers */
 				assert(streq(layers[i], "auto.sourcefree"));
-				continue;
+				plugin_log(rq->plugin, LOG_DBG, "Adding auto.sourcefree");
+				l = source_free_layer(layers, askrene->gossmap, source, localmods);
```

Comment:
So the previous version would first update localmods and after the loop it would apply `auto.sourcefree` if present.
Now we have to take care about the order of the layers again, because if `auto.localchans` comes after `auto.sourcefree` in the list then we will not be able to nullify the fees on private local channels, they don't "exist" yet in localmods.

Reply:
I didn't see the comment in the commit: "`getroutes` now applies `auto.sourcefree` layer in the order specified, so doesn't alter channels changed in later layers."